autotest/cluster/clusterx.py

```diff
@@ -0,0 +1,99 @@
+import time
+import traceback
+from typing import Any, Dict, Optional
+
+from clusterx.config import CLUSTER
+from clusterx.launcher import CLUSTER_MAPPING
+from clusterx.launcher.base import JobStatus
+
+
+class ClusterTaskExecutor:
+    def __init__(self):
+        cluster_spec = CLUSTER_MAPPING[CLUSTER]
+        cluster_cls = cluster_spec["type"]
+        params_cls = cluster_spec["params"]
+        cluster = cluster_cls()
+
+        self.cluster = cluster
+        self.params_cls = params_cls
+
+    def execute_task(self, task_config: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Suggest declaring a `TypedDict` or Pydantic `BaseModel` here for easier checking of config correctness.